plugins/inventory/vmware_vm_inventory.py

```diff
@@ -247,104 +280,193 @@ def check_requirements(self):
             raise AnsibleError("Missing one of the following : hostname, username, password. Please read "
                                "the documentation for more information.")
 
-    def _get_managed_objects_properties(self, vim_type, properties=None):
+    def get_managed_objects_properties(self, vim_type, properties=None, resources=None, strict=False):
         """
         Look up a Managed Object Reference in vCenter / ESXi Environment
         :param vim_type: Type of vim object e.g, for datacenter - vim.Datacenter
         :param properties: List of properties related to vim object e.g. Name
+        :param resources: List of resources to limit search scope
+        :param strict: Dictates if plugin raises error or just warns
         :return: local content object
         """
-        # Get Root Folder
-        root_folder = self.content.rootFolder
+        TraversalSpec = vmodl.query.PropertyCollector.TraversalSpec
+        FilterSpec = vmodl.query.PropertyCollector.FilterSpec
+        ObjectSpec = vmodl.query.PropertyCollector.ObjectSpec
+        PropertySpec = vmodl.query.PropertyCollector.PropertySpec
 
-        if properties is None:
-            properties = ['name']
+        resource_filters = resources or []
+        type_to_name_map = {}
 
-        # Create Container View with default root folder
-        mor = self.content.viewManager.CreateContainerView(root_folder, [vim_type], True)
-
-        # Create Traversal spec
-        traversal_spec = vmodl.query.PropertyCollector.TraversalSpec(
-            name="traversal_spec",
-            path='view',
-            skip=False,
-            type=vim.view.ContainerView
-        )
+        def _handle_error(message):
+            if strict:
+                raise AnsibleError(message)
```

Comment:
This strictness doesn't seem to be working like it appears it should. If you put in

```yaml
plugin: community.vmware.vmware_vm_inventory
properties:
- foefesfoienfgosefr
strict: true
```

then it still doesn't throw an error. I do not know why, maybe it's filtered out before here somehow?

Reply:
That is Ansible core behavior with inventory. Inventory error will be reported as "`[WARNING]:  * Failed to parse...` before it's try with other plugins and then give-up with `[WARNING]: Unable to parse...`. Then move to the next inventory if any exist. 

```
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are modifying the Ansible engine, or trying
out features under development. This is a rapidly changing source of code and can become unstable at any point.
[WARNING]:  * Failed to parse test-error.vmware.yaml with auto plugin: Invalid property name: fsakldjfalksdfjkl
[WARNING]:  * Failed to parse test_ansible_pr/test-error.vmware.yaml with yaml plugin: Plugin configuration YAML file, not YAML inventory
[WARNING]:  * Failed to parse test_ansible_pr/test-error.vmware.yaml with ini plugin: Invalid host pattern 'plugin:' supplied, ending in ':'
is not allowed, this character is reserved to provide a port.
[WARNING]: Unable to parse test_ansible_pr/test-error.vmware.yaml as an inventory source
[WARNING]: No inventory was parsed, only implicit localhost is available
{
    "_meta": {
        "hostvars": {}
    },
    "all": {
        "children": [
            "ungrouped"
        ]
    }
}
```

To see the traceback of the error, use "-vvv".


```sh
[WARNING]:  * Failed to parse test-error.vmware.yaml with auto plugin: Invalid property name: fsakldjfalksdfjkl
  File "ansible/lib/ansible/inventory/manager.py", line 287, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "ansible/lib/ansible/plugins/inventory/auto.py", line 58, in parse
    plugin.parse(inventory, loader, path, cache=cache)
  File "ansible_collections/community/vmware/plugins/inventory/vmware_vm_inventory.py", line 529, in parse
    cacheable_results = self._populate_from_source()
  File "collections/ansible_collections/community/vmware/plugins/inventory/vmware_vm_inventory.py", line 566, in _populate_from_source
    strict=self.get_option('strict')
  File "collections/ansible_collections/community/vmware/plugins/inventory/vmware_vm_inventory.py", line 391, in get_managed_objects_properties
    _handle_error("Invalid property name: %s" % err.name)
  File "collections/ansible_collections/community/vmware/plugins/inventory/vmware_vm_inventory.py", line 302, in _handle_error
    raise AnsibleError(message)
```

Reply:
No I really get results. If I put in some bogus properties and some real properties and run `ansible-inventory`, then I get hosts returned and the real properties inside of hostvars.

Like

```yaml
plugin: community.vmware.vmware_vm_inventory
hostnames:
- config.name
properties:
- name
- availableField
- config
- configIssue
strict: true
```

It gives hosts with `availableField`, among others.

I don't know exactly how painful this will be, because `compose` isn't implemented yet, and that's how I'm going to be constructing a lot of hostvars. But if anything doesn't match up, I need to get the errors from serializing the origin hostvars, and right now it appears that I can't do that.

I can't easily pinpoint _why_ this is the case from a quick read of the code. I would have to go do interactive debugging to figuring out why it's not looping over a property I expected it to serialize or fail on.